extension/agenthealth/handler/useragent/handler.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package useragent
+
+import (
+	"net/http"
+
+	"github.com/amazon-contributing/opentelemetry-collector-contrib/extension/awsmiddleware"
+	"go.uber.org/atomic"
+)
+
+type userAgentHandler struct {
+	userAgent          UserAgent
+	isUsageDataEnabled bool
+	header             *atomic.String
+}
+
+var _ awsmiddleware.RequestHandler = (*userAgentHandler)(nil)
+
+func (uah *userAgentHandler) ID() string {
+	return handlerID
+}
+
+func (uah *userAgentHandler) Position() awsmiddleware.HandlerPosition {
+	return awsmiddleware.After
+}
+
+// HandleRequest prepends the User-Agent header with the CloudWatch Agent's
+// user agent string.
+func (uah *userAgentHandler) HandleRequest(_ string, r *http.Request) {
```

Comment:
is there a future implementation of the string param?

Reply:
The user agent handler doesn't need it, but other ones, like the client stats one (measuring latency) do.